tests/agent_features/test_error_events.py

```diff
@@ -294,37 +301,171 @@ def test_error_event_outside_transaction_collect_error_events_false():
         notice_error(sys.exc_info(), application=app)
 
 
+_callback_called = threading.Event()
+_truncated_value = "A" * 300
+
+def error_group_callback(exc, data):
+    _callback_called.set()
+
+    if isinstance(exc, ValueError):
+        return "value"
+    elif isinstance(exc, ZeroDivisionError):
+        return _truncated_value
+    elif isinstance(exc, IndexError):
+        return []
+    elif isinstance(exc, TypeError):
+        return ""
+
+
+@pytest.mark.parametrize("exc_class,group_name,high_security", [
+    (ValueError, "value", False),
+    (ValueError, "value", True),
+    (TypeError, None, False),
+    (RuntimeError, None, False),
+    (IndexError, None, False),
+    (ZeroDivisionError, _truncated_value[:255], False),
+], ids=("standard", "high-security", "empty-string", "None-value", "bad-type", "truncated-value"))
 @reset_core_stats_engine()
-def test_error_callback_attributes():
+def test_error_group_name_callback(exc_class, group_name, high_security):
```

Comment:
Would it be worth adding another test where we call `set_error_group_callback()` twice to ensure the second call overwrites the first one?

Reply:
Done